### ISSUE TITLE
Create header above based on delimiter character

### ISSUE THREAD

**Issue by vinhtantran** [haozhu233/kableExtra#541]
There are multiple times I wanted to create a table with same column names but in different groups of columns. For example, when I want to show two models' coefficients side by side, so the top header is "Model 1" and "Model 2" while the next level headers are both "Estimate" and "SD". Data frames and tibbles don't allow columns with the same name so a way `gt` package is doing right now is to name your column "Model 1_Estimate" and "Model 2_Estimate" and then use `gt::tab_spanner_delim()` to split "_" into two headers. 

I guess there is some workaround but I really hope kable would have a similar feature.

**Comment by haozhu233** [comment]
Very nice suggestion! I just added a `header_separate` function to the github version. Please help me test it out. Here is a short example. 

```
data.frame(header = 0:2, a_p = 1:3, a_q = 2:4, b_p = 3:5, b_q = 4:6) %>%
  kable() %>%
  header_separate()
```

**Comment by vinhtantran** [comment]
Thank you for the quick effort. I tested the function and here are some of my suggestions:

1. The columns should be rearranged before performing splitting. It should also follow the order of appearance of columns, not alphabetically. For example, when my columns are Type, Coef, Log scale_Estimate, Dollar scale_Estimate, Log scale_SD, the resulting kable should be EMPTY/Type, EMPTY/Coef, Log scale/(Estimate, SD), Dollar scale/Estimate.

2. I tried adding a column named "Log_scale_SD", it created two level headers, one with Log, one with scale. I would suggest performing one level at a call, splitting at the first appearance of `sep` only.

I will think of more test cases.

**Comment by haozhu233** [comment]
@vinhtantran Thanks for your suggestions! Here are my feedbacks. 

1. "It should also follow the order of appearance of columns, not alphabetically" I actually hold a different view on this. Asking users to rearrange their columns thoughtfully using mature tools such as `dplyr::select` is a much cleaner workflow. At the same time, from the begining, I started kableExtra as a tool to "modifies" existing html/latex tables. It seems quite difficult on the code level to rearrange columns.

2. For the `header_separate` function, I borrowed the `separate` name and the `sep` option from `tidyr::separate`. In your case, I suggest using "Log scale_SD" and `header_separate(..., sep = "_")`
